package/MDAnalysis/topology/guessers.py

```diff
@@ -203,7 +200,8 @@ def guess_bonds(atoms, coords, **kwargs):
 
     lower_bound = kwargs.get('lower_bound', 0.1)
 
-    box = kwargs.get('box', None)
+    if box:
```

Comment:
This is bad and will typically fail for `box` being an array with *ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()*

You should use
```Python
if box is not None:
```
or similar.

Or do a `try/except`.

Make sure to assign `None` in all cases where there is no array. Downstream code might actually rely on `box = None` and not handle `box = False` or `box = 0` correctly.